recsys/inference/llm_ranking_predictor.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from langchain import PromptTemplate, LLMChain
+from langchain_openai import ChatOpenAI
+
+class Predict(object):
+    def __init__(self):
+        self.input_variables = ["age", "month_sin", "month_cos", "product_type_name", "product_group_name",
+                                "graphical_appearance_name", "colour_group_name", "perceived_colour_value_name",
+                                "perceived_colour_master_name", "department_name", "index_name", "index_group_name",
+                                "section_name", "garment_group_name"]
+        self.llm = self._build_lang_chain()
+
+    def predict(self, inputs):
+        logging.info(f"✅ Inputs: {inputs}")
+
+        # Extract ranking features and article IDs from the inputs
+        # limit to 20 candidates because of OPENAI token requests limitations
+        features = inputs[0].pop("ranking_features")[:20]
```

Comment:
As discussed, here we don't have only this problem but also that if we do inference for 100  candidates with the openai model it will result in very slow predict.  If our predictor takes more than 60 seconds to respond to the :predict request that the Hopsworks SDK does on the running deployment, it results in timeout and the socket on the machine running the Predictor is closed.